### DISCUSSION TITLE
How can I make <teleport> work with SSR?

### DISCUSSION THREAD

**Discussion by johnRivs** [frandiox/vite-ssr#90] [Q&A] [answered]
I was told the Teleport component should be able to be rendered during SSR. I was trying to use it for meta tags and couldn't get it to work in <head> but then I noticed my modals are not getting rendered in the server either. They're outside the component tree, as Vue expects.

Sorry if this doesn't belong here but I've asked in the Vue forums, Discord and Vite's repo with no luck :/

**Comment by frandiox** [comment, marked as answer]
I was trying to add support for teleports in SSR but found something that looks like a bug: https://github.com/vuejs/vue-next/issues/4440

Let's wait for some answers there.


>I was trying to use it for meta tags

Why don't you use `@vueuse/head` for meta tags? 🤔

**Reply by johnRivs** [reply]
Ahh good to know. Thanks.

Isn't stuff in `@vueuse` only for the composition API? I'm using options.

**Reply by frandiox** [reply]
You can probably use the component version: https://github.com/vueuse/head/blob/main/example/Contact.vue

**Reply by johnRivs** [reply]
Turns out I had it in my package.json cause I tried it a while ago. Doesn't work with SSR which I guess has to do with the teleport bug? Maybe they use the same technique.

I also saw this issue https://github.com/vueuse/head/issues/24

**Reply by frandiox** [reply]
It is working in the examples we have in this repo, though 🤔  At least with the composition API, I haven't tried with the options API :/

**Reply by johnRivs** [reply]
Oh I'm just dumb. I was going off the example in their repo instead of yours. I forgot to `return { head };` in the hook.

**Reply by frandiox** [reply]
ah lol, so does it work now?

**Reply by johnRivs** [reply]
Yeah, thanks once again 😅

**Comment by johnRivs** [comment]
Did https://github.com/vuejs/core/pull/5187 fix teleports working during SSR? Or is that only for async components? I'm stuck with hydration errors :/
